src/Symfony/Component/Form/Extension/Core/EventListener/ResizeFormListener.php

```diff
@@ -148,14 +149,16 @@ public function onSubmit(FormEvent $event)
             throw new UnexpectedTypeException($data, 'array or (\Traversable and \ArrayAccess)');
         }
 
-        if ($this->deleteEmpty) {
-            $previousData = $event->getForm()->getData();
+        if ($entryFilter = $this->deleteEmpty) {
+            $previousData = $form->getData();
             foreach ($form as $name => $child) {
+                /* @var $child FormInterface */
                 $isNew = !isset($previousData[$name]);
+                $isEmpty = $child->isEmpty() || (is_callable($entryFilter) && true === $entryFilter($child->getData()));
```

Comment:
I'm not convinced you need those standard checks to be run anyway. For instance, you'll already have to check for the data not being null inside your callable before further checks.